src/kernels/jupyter/types.ts

```diff
@@ -333,3 +336,12 @@ export interface IServerConnectionType {
     isLocalLaunch: boolean;
     onDidChange: Event<void>;
 }
+
+export const IRemoteKernelFinder = Symbol('IRemoteKernelFinder');
+export interface IRemoteKernelFinder extends IContributedKernelFinder {
```

Comment:
I would suggest creating two seprate interfaces, one to list kernel specs and one to list the live kernel connectiosn (active sessions).
The kernel sessions is what can change over time, we don't need to re-query the kernel specs everytime we need new kernel sessions.

Reply:
@DonJayamanne yes we can do that. This PR is the first part of moving towards contribution model but not necessarily introducing new interfaces, that can come in follow up PRs.

Reply:
Let's book sometime to discuss the separation though, it would be more productive (for me ;) ) if I have you look over my shoulder.